src/wp-includes/option.php

```diff
@@ -2893,3 +2879,50 @@ function filter_default_option( $default_value, $option, $passed_default ) {
 
 	return $registered[ $option ]['default'];
 }
+
+/**
+ * Check if two database values are equal.
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param mixed $old_value The old value to compare.
+ * @param mixed $new_value The new value to compare.
+ * @return bool True if the values are equal, false otherwise.
+ */
+function is_equal_database_value( $old_value, $new_value ) {
+	$values = array(
```

Comment:
Wonder if we could use `compact` here?

Reply:
I don't think there's any value in using `compact`. The way it is written is easier to understand and not slower.